invenio_mail/ext.py

```diff
@@ -67,5 +67,11 @@ def init_config(self, app):
     def print_email(self, message, app):
         """Print mail to stream."""
         with self._lock:
-            self.stream.write("{0}\n{1}\n".format(message.as_string(), '-'*79))
-            self.stream.flush()
+            invenio_mail = app.extensions['invenio-mail']
```

Comment:
I don't really follow this change. You are using `self._lock` which is possibly on different `invenio-mail` instance.

a) revert the change
b) create static method and use `invenio_mail._lock` instead of `self._lock`
c) propose different change